pkg/apis/v1/nodepool_conversion_test.go

```diff
@@ -537,4 +556,27 @@ var _ = Describe("Convert V1beta1 to V1 NodePool API", func() {
 			Expect(v1beta1nodepool.Status.Resources[resource]).To(Equal(v1nodepool.Status.Resources[resource]))
 		}
 	})
+	Context("Round Trip", func() {
+		DescribeTable(
+			"NillableDuration",
+			func(value string, field func() *v1beta1.NillableDuration) {
+				str := fmt.Sprintf("%q", value)
+				duration := v1beta1.NillableDuration{}
+				Expect(json.Unmarshal([]byte(str), &duration)).Should(Succeed())
+				*field() = duration
+				Expect(v1nodepool.ConvertFrom(ctx, v1beta1nodepool)).To(Succeed())
+				Expect(v1nodepool.ConvertTo(ctx, v1beta1nodepool)).To(Succeed())
+				result, err := json.Marshal(*field())
+				Expect(err).To(BeNil())
+				Expect(string(result)).To(Equal(str))
+			},
+			Entry("spec.template.spec.expireAfter", "720h", func() *v1beta1.NillableDuration { return &v1beta1nodepool.Spec.Disruption.ExpireAfter }),
+			Entry("spec.disruption.consolidateAfter", "15m", func() *v1beta1.NillableDuration {
+				if v1beta1nodepool.Spec.Disruption.ConsolidateAfter == nil {
```

Comment:
What about for WhenEmpty?

Reply:
This didn't have to do with the consolidation type, it was just a hack since I was mutating the field through a pointer and needed to ensure it was non-nil. I've refactored the test to remove this.